models/astrocyte_lr_1994.cpp

```diff
@@ -0,0 +1,533 @@
+  {
+    throw BadProperty( "Calcium concentration must be non-negative." );
+  }
+  if ( y_[ h_IP3R ] < 0 || y_[ h_IP3R ] > 1 )
+  {
+    throw BadProperty( "The fraction of active IP3 receptors on the astrocytic ER must be between 0 and 1." );
+  }
+}
+
+nest::astrocyte_lr_1994::Buffers_::Buffers_( astrocyte_lr_1994& n )
+  : logger_( n )
+  , s_( nullptr )
+  , c_( nullptr )
+  , e_( nullptr )
+{
+  // Initialization of the remaining members is deferred to
+  // init_buffers_().
+}
+
+nest::astrocyte_lr_1994::Buffers_::Buffers_( const Buffers_&, astrocyte_lr_1994& n )
+  : logger_( n )
+  , s_( nullptr )
+  , c_( nullptr )
+  , e_( nullptr )
+{
+  // Initialization of the remaining members is deferred to
+  // init_buffers_().
+}
+
+/* ----------------------------------------------------------------
+ * Default and copy constructor for node, and destructor
+ * ---------------------------------------------------------------- */
+
+nest::astrocyte_lr_1994::astrocyte_lr_1994()
+  : ArchivingNode()
+  , P_()
+  , S_( P_ )
+  , B_( *this )
+{
+  recordablesMap_.create();
+}
+
+nest::astrocyte_lr_1994::astrocyte_lr_1994( const astrocyte_lr_1994& n )
+  : ArchivingNode( n )
+  , P_( n.P_ )
+  , S_( n.S_ )
+  , B_( n.B_, *this )
+{
+}
+
+nest::astrocyte_lr_1994::~astrocyte_lr_1994()
+{
+  // GSL structs may not have been allocated, so we need to protect destruction
+  if ( B_.s_ )
+  {
+    gsl_odeiv_step_free( B_.s_ );
+  }
+  if ( B_.c_ )
+  {
+    gsl_odeiv_control_free( B_.c_ );
+  }
+  if ( B_.e_ )
+  {
+    gsl_odeiv_evolve_free( B_.e_ );
+  }
+}
+
+/* ----------------------------------------------------------------
+ * Node initialization functions
+ * ---------------------------------------------------------------- */
+
+void
+nest::astrocyte_lr_1994::init_buffers_()
+{
+  B_.spike_exc_.clear(); // includes resize
+  B_.currents_.clear();
+  B_.sic_values.resize(
+    kernel().connection_manager.get_min_delay(), 0.0 ); // set size of SIC buffer according to min_delay
+
+  ArchivingNode::clear_history();
+
+  B_.logger_.reset();
+
+  B_.step_ = Time::get_resolution().get_ms();
+  B_.IntegrationStep_ = B_.step_;
+
+  if ( not B_.s_ )
+  {
+    B_.s_ = gsl_odeiv_step_alloc( gsl_odeiv_step_rkf45, State_::STATE_VEC_SIZE );
+  }
+  else
+  {
+    gsl_odeiv_step_reset( B_.s_ );
+  }
+
+  if ( not B_.c_ )
+  {
+    B_.c_ = gsl_odeiv_control_y_new( 1e-3, 0.0 );
+  }
+  else
+  {
+    gsl_odeiv_control_init( B_.c_, 1e-3, 0.0, 1.0, 0.0 );
+  }
+
+  if ( not B_.e_ )
+  {
+    B_.e_ = gsl_odeiv_evolve_alloc( State_::STATE_VEC_SIZE );
+  }
+  else
+  {
+    gsl_odeiv_evolve_reset( B_.e_ );
+  }
+
+  B_.sys_.function = astrocyte_lr_1994_dynamics;
+  B_.sys_.jacobian = nullptr;
+  B_.sys_.dimension = State_::STATE_VEC_SIZE;
+  B_.sys_.params = reinterpret_cast< void* >( this );
+
+  B_.J_noise_ = 0.0;
+}
+
+void
+nest::astrocyte_lr_1994::pre_run_hook()
+{
+  // ensures initialization in case mm connected after Simulate
+  B_.logger_.init();
```

Comment:
The variable `max_Ca_` is not initialized here.

Reply:
Hi, it turned out this variable is not required. I have already deleted it. Sorry for the confusion.